icu4c/source/test/intltest/unitstest.cpp

```diff
@@ -160,4 +166,102 @@ void UnitsTest::testArea() {
     }
 }
 
+/**
+ * Returns a StringPiece pointing at the given field with space prefixes and
+ * postfixes trimmed off.
+ */
+StringPiece trimField(char *(&field)[2]) {
+  char *start = field[0];
+  while (start < field[1] && (start[0]) == ' ') {
+    start++;
+  }
+  int32_t length = (int32_t)(field[1] - start);
+  while (length > 0 && (start[length - 1]) == ' ') {
+    length--;
+  }
+  return StringPiece(start, length);
+}
+
+/**
+ * WIP(hugovdm): deals with a single data-driven unit test for unit conversions.
+ * This is a UParseLineFn as required by u_parseDelimitedFile.
+ */
+static void U_CALLCONV unitsTestDataLineFn(void *context, char *fields[][2],
+                                           int32_t fieldCount,
+                                           UErrorCode *pErrorCode) {
+  (void)fieldCount; // unused UParseLineFn variable
+  IcuTestErrorCode status(*(UnitsTest *)context, "unitsTestDatalineFn");
+
+  StringPiece quantity = trimField(fields[0]);
+  StringPiece x = trimField(fields[1]);
+  StringPiece y = trimField(fields[2]);
+  StringPiece commentConversionFormula = trimField(fields[3]);
+  StringPiece utf8Expected = trimField(fields[4]);
+
+  UNumberFormat *nf =
+      unum_open(UNUM_DEFAULT, NULL, -1, "en_US", NULL, pErrorCode);
+  UnicodeString uExpected = UnicodeString::fromUTF8(utf8Expected);
+  double expected = unum_parseDouble(nf, uExpected.getBuffer(),
+                                     uExpected.length(), 0, pErrorCode);
+  unum_close(nf);
```

Comment:
Technically this works, but please use the following to convert a string to a double…

```cpp
DecimalQuantity expected;
expected.setToDecNumber({uExpected.getBuffer(), uExpected.length()}, *pErrorCode);
double expectedDouble = expected.toDouble();
```

Reply:
I'm currently still parsing "1,000.00". When I do this with DecimalQuantity, it gives U_DECIMAL_NUMBER_SYNTAX_ERROR. Is it better to use DecimalQuantity and manually strip out the thousands separators? It seemed reasonable to me to simply treat "1,000.00" as a localised number format.

Reply:
OK.  Fine to leave it as-is.